### PR TITLE
Support peer with same IP but different port

### PR DESCRIPTION
**Pull request by Suraiya-Hameed**
This PR is to differentiate between peers with same IP but different ports.

- Peer key will now append port number if it exists `.../peer_v4/x.x.x.x-<port>`
- If listenPort is updated, then update the peers. If we have calico/node as peer, we might be using listenPort to peer with it, so update all the peers if listenPort is updated.

projectcalico/confd#341 (comment)

### CODE REVIEW

File: etc/calico/confd/templates/bird.cfg.template [hunk 1]
```diff
@@ -166,6 +166,9 @@ template bgp bgp_template {
 {{if ls "/global/peer_v4"}}
 {{range gets "/global/peer_v4/*"}}{{$data := json .Value}}
 {{$nums := split $data.ip "."}}{{$id := join $nums "_"}}
+{{- if $data.port}}
+{{- $id = printf "%s_%.0f" $id $data.port}}
```
Comment (fasaxc) [inline etc/calico/confd/templates/bird.cfg.template:170]: IS `%d` (for decimal) not available; seems odd to use a float format

File: etc/calico/confd/templates/bird6.cfg.template [hunk 2]
```diff
@@ -194,6 +197,9 @@ protocol bgp Global_{{$id}} from bgp_template {
 {{if ls $node_peers_key}}
 {{range gets (printf "%s/*" $node_peers_key)}}{{$data := json .Value}}
 {{$nums := split $data.ip ":"}}{{$id := join $nums "_"}}
+{{- if $data.port}}
+{{- $id = printf "%s_%.0f" $id $data.port}}
```
Comment (fasaxc) [inline etc/calico/confd/templates/bird6.cfg.template:201]: Could this algorithm give collisions between IDs for IPv6 peers with no port and ones with a port?  I'm worried that the colon of the IPv6 address gets converted to `_` too so maybe there's a way to make them collide.  Quick fix would be to do ```
$id = printf "%s_port_%.0f" $id $data.port
```
so it's totally unambiguous.

File: etc/calico/confd/templates/bird.cfg.template (hunk 1 above)
Comment (Suraiya-Hameed) [reply etc/calico/confd/templates/bird.cfg.template:170]: `%d` was the initial choice, say if port is 179 it prints `%!d(float64=179)`, that's reason for using `%f`

File: etc/calico/confd/templates/bird6.cfg.template (hunk 2 above)
Comment (Suraiya-Hameed) [reply etc/calico/confd/templates/bird6.cfg.template:201]: yes, that will be a good way to differentiate between port and IP, I'll update it for both IPv6 and IPv4

**Pull request merged**